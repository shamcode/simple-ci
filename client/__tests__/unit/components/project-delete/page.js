import { createDI } from 'sham-ui';
import Page from '../../../../src/components/project-delete/page.sfc';
import * as directives from 'sham-ui-directives';
import hrefto from 'sham-ui-router/lib/href-to';
import renderer from 'sham-ui-test-helpers';

const flushPromises = () => new Promise( resolve => setImmediate( resolve ) );

it( 'renders correctly', () => {
    const DI = createDI();
    const storeMock = jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValue( '/' )
    } );

    const meta = renderer( Page, {}, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( storeMock.mock.calls[ 0 ][ 0 ] ).toBe( 1 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'render errors', async() => {
    expect.assertions( 1 );
    const DI = createDI();

    const storeMock = jest.fn().mockReturnValue( Promise.reject() );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        storage: { params: {} },
        generate: jest.fn().mockReturnValue( '/' )
    } );

    const meta = renderer( Page, {}, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    await flushPromises();

    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'delete fail', async() => {
    expect.assertions( 4 );

    const DI = createDI();
    const deleteProject = jest.fn();
    DI.bind( 'store', {
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( {
            project: {
                id: 1,
                name: 'Test',
                cwd: '/tmp/'
            }
        } ) ),
        deleteProject: deleteProject
    } );
    const navigateMock = jest.fn();
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValue( '/' ),
        navigate: navigateMock
    } );

    const meta = renderer( Page, {}, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    const { ctx: { container } } = meta;
    await flushPromises();

    deleteProject.mockReturnValueOnce( Promise.reject() );
    container.querySelector( '.ok' ).click();
    await flushPromises();

    expect( deleteProject.mock.calls ).toHaveLength( 1 );
    expect( navigateMock.mock.calls ).toHaveLength( 1 );
    expect( navigateMock.mock.calls[ 0 ][ 0 ] ).toBe( '/' );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'cancel', async() => {
    expect.assertions( 2 );

    const DI = createDI();
    DI.bind( 'store', {
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( {
            project: {
                id: 1,
                name: 'Test',
                cwd: '/tmp/'
            }
        } ) )
    } );
    const navigateMock = jest.fn();
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValue( '/' ),
        navigate: navigateMock
    } );

    const meta = renderer( Page, {}, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    const { ctx: { container } } = meta;
    await flushPromises();

    container.querySelector( '.cancel' ).click();
    expect( navigateMock.mock.calls ).toHaveLength( 1 );
    expect( navigateMock.mock.calls[ 0 ][ 0 ] ).toBe( '/' );

    await flushPromises();
} );
