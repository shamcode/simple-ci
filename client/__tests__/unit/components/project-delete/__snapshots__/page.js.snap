// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete fail 1`] = `
Object {
  "Constructor": "ProjectDeletePage",
  "Options": Object {
    "ID": "component",
    "bodyText": [Function],
    "cancelDeleteProject": [Function],
    "containerSelector": "body",
    "dataLoaded": true,
    "dataSaving": true,
    "deleteProject": [Function],
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [],
    "project": Object {
      "cwd": "/tmp/",
      "id": 1,
      "name": "Test",
    },
    "types": Array [],
  },
  "html": "
<ul class=\\"breadcrumb\\">
  <li>
    <a href=\\"/\\"> Projects </a>
  </li>
  <li>
    <a href=\\"/\\">Test</a>
  </li>
  <!--if-->
  <!--if-->
  <li>Delete</li>
</ul>
<!--BreadCrumb-->
<!--breadcrumbs-->
<!--if-->
<!--ErrorsPanel-->
<div class=\\"modal\\">
  <div class=\\"content\\">
    <div class=\\"header\\">Delete project</div>
    <div class=\\"body\\">Do you want delete project \\"Test\\"?</div>
    <div class=\\"footer\\">
      <button class=\\"ok danger\\" disabled=\\"\\"><span>Confirm</span></button>
      <button class=\\"cancel\\"><span>Cancel</span></button>
    </div>
  </div>
</div>
<!--ConfirmModal-->
<!--if-->
<!--content-->
<!--if-->
<!--PageContent-->
",
}
`;

exports[`render errors 1`] = `
Object {
  "Constructor": "ProjectDeletePage",
  "Options": Object {
    "ID": "component",
    "bodyText": [Function],
    "cancelDeleteProject": [Function],
    "containerSelector": "body",
    "dataLoaded": true,
    "dataSaving": false,
    "deleteProject": [Function],
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [
      "Load project fail!",
    ],
    "project": Object {
      "chains": Array [],
    },
    "types": Array [],
  },
  "html": "
<ul class=\\"breadcrumb\\">
  <li>
    <a href=\\"/\\"> Projects </a>
  </li>
  <!--if-->
  <li></li>
</ul>
<!--BreadCrumb-->
<!--breadcrumbs-->
<ul class=\\"errors-panel\\">
  <li>Load project fail!</li>
</ul>
<!--if-->
<!--ErrorsPanel-->
<!--if-->
<!--content-->
<!--if-->
<!--PageContent-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Constructor": "ProjectDeletePage",
  "Options": Object {
    "ID": "component",
    "bodyText": [Function],
    "cancelDeleteProject": [Function],
    "containerSelector": "body",
    "dataLoaded": false,
    "dataSaving": false,
    "deleteProject": [Function],
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [],
    "project": Object {},
    "types": Array [],
  },
  "html": "
<ul class=\\"breadcrumb\\">
  <li>
    <a href=\\"/\\"> Projects </a>
  </li>
  <!--if-->
  <li></li>
</ul>
<!--BreadCrumb-->
<!--breadcrumbs-->
<!--if-->
<!--FlushPageDataLoading-->
<!--if-->
<!--PageContent-->
",
}
`;
